aten/src/ATen/native/cuda/Math.cuh

```diff
@@ -0,0 +1,87 @@
+#pragma once
+#include <type_traits>
+
+namespace at {
+namespace native {
+
+/*
+* The following function was converted to CUDA form from code that comes
+* with the following copyright notice. It has been released under the BSD license.
+*
+* Cephes Math Library Release 2.8:  June, 2000
+* Copyright 1984, 1987, 1992, 2000 by Stephen L. Moshier
+*/
+template <typename scalar_t>
+static inline __host__ __device__ scalar_t calc_digamma(scalar_t x) {
```

Comment:
One PyTorch convention that I've noticed in the codebase is that we generally name the type as `accscalar_t` to represent that it's the type we are accumulating into. Do you think it would be reasonable to rename this type here?

Reply:
(Resolved offline): we're going to use `at::acc_type`: https://github.com/pytorch/pytorch/blob/df043cd49d718c144ca719e593c69113ae60371d/aten/src/ATen/native/cuda/FractionalMaxPool2d.cu#L44

Reply:
changed to accscalar_t

Reply:
(Just trying to keep acc_type properly used across the codebase...)

I'm a bit skeptic about using acc_type here. I opened a PR trying to explain what acc_type means: #25732. If we only want to promote half to float on CUDA, I think that is for a very different reason, because half is slower than float on many GPU models, not merely for accumulation purpose.